### PR TITLE
feat(redux):setUp[#187300086]

### PR DESCRIPTION
**Pull request by Mukunzijames**
What does this PR do?
configure redux toolkit for next js14

Description of Task to be completed?

Configure The App to follow rules

Configure redux toolkit and slices 

How should this be manually tested?

to run dev use "npm run dev "

redux toolkit automatically runs it self whenever you push

Any background context you want to provide?
-this is for specifically allow order of codebase

What are the relevant pivotal tracker/Trello stories?

id of the project

#187300086

### CODE REVIEW

**Review by musabehonore — APPROVED** [review]

**Review by fab-ryan — CHANGES_REQUESTED** [review]
any type are not allowed,
create custom hook for getting typed actions and typed slice

**Comment by Mukunzijames** [comment]
> any type are not allowed, create custom hook for getting typed actions and typed slice

solved

**Review by IRASUBIZA-Elyse — APPROVED** [review]

**Review by Nadinefiona — APPROVED** [review]

**Review by fab-ryan — CHANGES_REQUESTED** [review]
make the following structure folder for hooks
- slices folders not actionType folder should be there

File: src/redux/provider.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use client'
+import { Provider } from "react-redux";
+import  store  from "./store";
+
+function Providers({children}: {children: React.ReactNode}) {
+    return <Provider store={store}>{children}</Provider>;
+}
+
+export default Providers;
```
Comment (fab-ryan) [inline src/redux/provider.tsx]: new line here

File: src/redux/slices/postSlice.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+interface PostsState {
+  posts: Post[];
+}
+
+const initialState: PostsState = {
+  posts: [],
+};
+
```
Comment (fab-ryan) [inline src/redux/slices/postSlice.ts:17]: where is that function

**Review by fab-ryan — CHANGES_REQUESTED** [review]
you have removed the previous

File: src/redux/slices/postThunk.ts
Comment (fab-ryan) [inline src/redux/slices/postThunk.ts:1]: don't create separate function please

File: src/redux/slices/postSlice.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { addPost, deletePost } from '../actions/postActions';
+import { Post } from '../slices/postThunk';
+
```
Comment (fab-ryan) [inline src/redux/slices/postSlice.ts:9]: add it here

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Please follow my requests

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**